engine-adapter/camunda-platform-7-remote-core/src/main/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/remote/task/delivery/subscribe/SubscribingClientServiceTaskDelivery.kt

```diff
@@ -14,55 +16,65 @@ import org.camunda.bpm.client.topic.TopicSubscriptionBuilder
 class SubscribingClientServiceTaskDelivery(
   private val externalTaskClient: ExternalTaskClient,
   private val subscriptionRepository: SubscriptionRepository,
-  private val lockDuration: Long
-) {
+  private val lockDuration: Long,
+  private val retryTimeout: Long,
+  private val retries: Int,
+  ) {
+
+  companion object: KLogging()
 
   fun subscribe() {
-    subscriptionRepository.getTaskSubscriptions()
-      .forEach { subscription ->
-        // this is a job to subscribe to.
-        externalTaskClient
-          .subscribe(subscription.taskDescriptionKey)
-          .lockDuration(lockDuration)
-          .handler { externalTask, externalTaskService ->
-            if (subscription.matches(externalTask)) {
-              subscriptionRepository.activateSubscriptionForTask(externalTask.id, subscription)
 
-              val variables = if (subscription.payloadDescription == null) {
-                externalTask.allVariables
-              } else {
-                if (subscription.payloadDescription!!.isEmpty()) {
-                  mapOf()
-                } else {
-                  externalTask.allVariables.filter { subscription.payloadDescription!!.contains(it.key) }
-                }
-              }
+    val subscriptions = subscriptionRepository.getTaskSubscriptions()
+    if (subscriptions.isNotEmpty()) {
+      logger.trace { "PROCESS-ENGINE-C7-REMOTE-030: subscribing to external tasks for: $subscriptions" }
+
+      subscriptions
+        .forEach { subscription ->
+          // this is a job to subscribe to.
+          externalTaskClient
+            .subscribe(subscription.taskDescriptionKey)
+            .lockDuration(lockDuration)
+            .handler { externalTask, externalTaskService ->
+              if (subscription.matches(externalTask)) {
```

Comment:
Just a idea, but on my screen this lambda is already bigger than one page ... should we extract it maybe?